br/pkg/restore/misc.go

```diff
@@ -44,6 +53,182 @@ const (
 	CoarseGrained Granularity = "coarse-grained"
 )
 
+const logRestoreTableIDBlocklistFilePrefix = "v1/log_restore_tables_blocklists"
+
+type LogRestoreTableIDsBlocklistFile struct {
+	// RestoreCommitTs records the timestamp after PITR restore done. Only the later PITR restore from the log backup of the cluster,
+	// whose BackupTS is not less than it, can ignore the restore table IDs blocklist recorded in the file.
+	RestoreCommitTs uint64 `protobuf:"varint,1,opt,name=restore_commit_ts,proto3"`
+	// SnapshotBackupTs records the BackupTS of the PITR restore. Any PITR restore from the log backup of the cluster, whose restoredTS
+	// is less than it, can ignore the restore table IDs blocklist recorded in the file.
+	SnapshotBackupTs uint64 `protobuf:"varint,2,opt,name=snapshot_backup_ts,proto3"`
+	// TableIDs records the table IDs blocklist of the cluster running the log backup task.
+	TableIds []int64 `protobuf:"varint,3,rep,packed,name=table_ids,proto3"`
+	// Checksum records the checksum of other fields.
+	Checksum []byte `protobuf:"bytes,4,opt,name=checksum,proto3"`
+}
+
+func (m *LogRestoreTableIDsBlocklistFile) Reset()         { *m = LogRestoreTableIDsBlocklistFile{} }
+func (m *LogRestoreTableIDsBlocklistFile) String() string { return proto.CompactTextString(m) }
+func (m *LogRestoreTableIDsBlocklistFile) ProtoMessage()  {}
+
+func (m *LogRestoreTableIDsBlocklistFile) filename() string {
+	return fmt.Sprintf("%s/R%016X_S%016X.meta", logRestoreTableIDBlocklistFilePrefix, m.RestoreCommitTs, m.SnapshotBackupTs)
+}
+
+func parseLogRestoreTableIDsBlocklistFileName(filename string) (restoreCommitTs, snapshotBackupTs uint64, parsed bool) {
+	filename = path.Base(filename)
+	if !strings.HasSuffix(filename, ".meta") {
+		return 0, 0, false
+	}
+	if filename[0] != 'R' {
+		return 0, 0, false
+	}
+	ts, err := strconv.ParseUint(filename[1:17], 16, 64)
```

Comment:
is it possible to not use the absolute high/low bounds here? since hex digits is 0 to f, separator R,S OR _ are not included